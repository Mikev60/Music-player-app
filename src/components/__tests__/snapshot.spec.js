import { test, expect, describe } from 'vitest'
import SongItem from '@/components/SongItem.vue'
import { shallowMount, RouterLinkStub } from '@vue/test-utils' //Shallow pour éviter que tous les children soient montés

describe('Snapshots SongItem.vue', () => {
  test('Renders correctly', () => {
    const song = {
      docID: 'abc',
      modified_name: 'test',
      display_name: 'test',
      comment_count: 5
    }
    const wrapper = shallowMount(SongItem, {
      propsData: { song },
      global: {
        components: {
          'router-link': RouterLinkStub
        }
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
