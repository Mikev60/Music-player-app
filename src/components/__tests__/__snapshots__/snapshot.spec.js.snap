// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshots SongItem.vue > Renders correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
  id="song-id-abc"
>
  <div>
    <router-link-stub
      class="font-bold block text-gray-600"
      custom="false"
      href="#"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
    >
      test
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <router-link-stub
      class="comments"
      custom="true"
      to="[object Object]"
    />
  </div>
</li>
`;
